src/Config/Option/ArrayOption.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Config\Option;
+
+use Glpi\Application\View\TemplateRenderer;
+use Glpi\Config\ConfigOption;
+use Glpi\Config\ConfigScope;
+use Glpi\Config\ConfigSection;
+use Glpi\Config\InputType;
+
+class ArrayOption extends ConfigOption
+{
+    public function __construct(
+        array $scopes,
+        ConfigSection $section,
+        string $name,
+        string $label,
+        array $type_options = [],
```

Comment:
We should try to ban usage of array of options on new code we produce. It is often hard to know what are the expected keys and then, as they are often passed from methods to method, it is hard to know how they are really used.
Also, we cannot benefit of checks induced by usage of strict types and tools like PHPStan cannot really help us, unless we provide some specific PHPDoc.
Using dedicated arguments is also easy to use, now that we can name arguments in PHP (e.g. `new Obj(content: $something, option_a: $a, xyz: true)`.

For the current PR, I also think we should all passing any rendering option to the `Glpi\Config\Option\XXXOption`  classes as, I think, a same type of option should always be rendered the same way. If there is a specific need for one option, we should probably create a dedicated class.

On this specific class, I guess the only fields options that are to be know are the list of available values, and whether user can select multiple values (it could even be two disting classes to handle this).